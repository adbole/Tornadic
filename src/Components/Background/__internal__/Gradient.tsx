import React from "react";
import styled from "@emotion/styled";

import type WeatherCondition from "ts/WeatherCondition";


const Cover = styled.div({
    position: "fixed",
    top: 0,
    left: 0,

    width: "100%",
    height: "100%",

    zIndex: -1,
});

export default function Gradient({
    isDay,
    condition,
}: {
    isDay: boolean;
    condition: WeatherCondition["type"];
}) {
    const [from, to] = React.useMemo(() => {
        switch (condition) {
            case "Overcast":
                return isDay ? ["#acb7bd", "#73bae1"] : ["#2d3438", "#1a1b1c"];
            case "Rain":
            case "Drizzle":
            case "Freezing Drizzle":
            case "Freezing Rain":
            case "Rain Showers":
                return ["#0f1c50", "#8da3bd"];
            case "Thunderstorms":
            case "Thunderstorms and Hail":
                return ["#4f5158", "#8da3bd"];
            case "Snow":
            case "Snow Grains":
            case "Snow Showers":
                return ["#797c87", "#acb7bd"];
            default:
                return isDay ? ["#7a98b2", "#bfc5c6"] : ["#151618", "#18202d"];
        }
    }, [isDay, condition]);

    const background = `linear-gradient(to bottom, ${from}, ${to})`;

    return <Cover style={{ background }} data-test-style-background={background} />;
}
