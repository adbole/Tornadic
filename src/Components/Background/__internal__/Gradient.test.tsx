import { render } from "@testing-library/react";

import type WeatherCondition from "ts/WeatherCondition";

import Gradient from "./Gradient";



describe.each([
    ["Day", true],
    ["Night", false],
])("%s", (_, isDay) => {
    test.each([
        "Clear",
        "Drizzle",
        "Foggy",
        "Freezing Drizzle",
        "Freezing Rain",
        "Mostly Clear",
        "Overcast",
        "Partly Cloudy",
        "Rain",
        "Rain Showers",
        "Snow",
        "Snow Grains",
        "Snow Showers",
        "Thunderstorms",
        "Thunderstorms and Hail",
    ] as WeatherCondition["type"][])("%s", condition => {
        const { container } = render(<Gradient isDay={isDay} condition={condition} />);

        expect(container).toMatchSnapshot();
    });
});
