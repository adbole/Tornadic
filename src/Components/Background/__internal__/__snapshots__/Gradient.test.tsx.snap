// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Day > Clear 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #7a98b2, #bfc5c6)"
  />
</div>
`;

exports[`Day > Drizzle 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Day > Foggy 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #7a98b2, #bfc5c6)"
  />
</div>
`;

exports[`Day > Freezing Drizzle 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Day > Freezing Rain 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Day > Mostly Clear 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #7a98b2, #bfc5c6)"
  />
</div>
`;

exports[`Day > Overcast 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #acb7bd, #73bae1)"
  />
</div>
`;

exports[`Day > Partly Cloudy 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #7a98b2, #bfc5c6)"
  />
</div>
`;

exports[`Day > Rain 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Day > Rain Showers 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Day > Snow 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Day > Snow Grains 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Day > Snow Showers 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Day > Thunderstorms 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #4f5158, #8da3bd)"
  />
</div>
`;

exports[`Day > Thunderstorms and Hail 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #4f5158, #8da3bd)"
  />
</div>
`;

exports[`Night > Clear 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #151618, #18202d)"
  />
</div>
`;

exports[`Night > Drizzle 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Night > Foggy 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #151618, #18202d)"
  />
</div>
`;

exports[`Night > Freezing Drizzle 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Night > Freezing Rain 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Night > Mostly Clear 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #151618, #18202d)"
  />
</div>
`;

exports[`Night > Overcast 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #2d3438, #1a1b1c)"
  />
</div>
`;

exports[`Night > Partly Cloudy 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #151618, #18202d)"
  />
</div>
`;

exports[`Night > Rain 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Night > Rain Showers 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #0f1c50, #8da3bd)"
  />
</div>
`;

exports[`Night > Snow 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Night > Snow Grains 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Night > Snow Showers 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #797c87, #acb7bd)"
  />
</div>
`;

exports[`Night > Thunderstorms 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #4f5158, #8da3bd)"
  />
</div>
`;

exports[`Night > Thunderstorms and Hail 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-test-style-background="linear-gradient(to bottom, #4f5158, #8da3bd)"
  />
</div>
`;
